javascript/filter.js

```diff
@@ -0,0 +1,42 @@
+function filterSelection(section) {
+  let element = document.getElementsByClassName("portfolio__box");
+  if (section == "all") {
+    section = "";
+  }
+  for (let i = 0; i < element.length; i++) {
+    RemoveClass(element[i], "portfolio__show");
+    if (element[i].className.indexOf(section) > -1)
+      AddClass(element[i], "portfolio__show");
+  }
+}
+
+function AddClass(element, name) {
+  let arr1 = element.className.split(" ");
+  let arr2 = name.split(" ");
+  for (let i = 0; i < arr2.length; i++) {
+    if (arr1.indexOf(arr2[i]) == -1) {
+      element.className += " " + arr2[i];
+    }
+  }
+}
+
+function RemoveClass(element, name) {
+  let arr1 = element.className.split(" ");
+  let arr2 = name.split(" ");
+  for (let i = 0; i < arr2.length; i++) {
+    while (arr1.indexOf(arr2[i]) > -1) {
+      arr1.splice(arr1.indexOf(arr2[i]), 1);
+    }
+  }
+  element.className = arr1.join(" ");
+}
```

Comment:
Actually good job for pulling it off in such a hard way 😃, but you could have done it in a much easier way. Look into [classList.add, classList.remove](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList).

In my response above I left your code on purpose so you can try to simplify it yourself. If you have any question message me on Slack.